#557 项目初始化时bean重复加载

Zhazhanan opened the pull request:
### What is this PR for?
A few sentences describing the overall goals of the pull request's commits.
修复项目初始化时bean重复加载


### What type of PR is it?
[Bug Fix | Improvement | Feature | Documentation | Hot Fix | Refactoring]
Improvement

### Todos
问题定位：WebConfig.java配置和Spring xml配置重复
解决方法：去掉javaconfig配置方式。1：开启springmvc、mvc的方法处理适配器等配置放在spring-mvc.xml；2：扫描包、加载属性文件、aop等配置放在spring.xml；

### What is the Github issue?
https://github.com/TuiQiao/CBoard/issues/557

### How should this be tested?
Outline the steps to test the PR here.
1.项目正常启动，bean未重复加载
2.页面正常访问
3.AOP未测试

### Screenshots (if appropriate)
无

### Questions:
* Is there breaking changes for older versions?
修改配置文件
* Does this needs documentation?
不需要

Commit:
#557 项目初始化时bean重复加载
问题定位：WebConfig.java配置和Spring xml配置重复
解决方法：去掉javaconfig配置方式。1：开启springmvc、mvc的方法处理适配器等配置放在spring-mvc.xml；2：扫描包、加载属性文件、aop等配置放在spring.xml；

yzhang921 commented:
反过来把xml里面的mvc配置全部用注解实现会更加方便后续的配置管理, 比如过swager, 虽然我几次尝试加入都以失败告终 😢  
另外, 删除的文件可以直接删掉, 一堆废弃的注释放上面反而会影响阅读

Zhazhanan commented:
@yzhang921 你说的没错，我看到xml与javaconfig在混用，所以我统一了一下，如果有需要，我可以去做全注解现实，顺便加上swagger2

yzhang921 commented:
可以试试

Commit:
javaconfig替代xml配置
修改位置：
1、去除spring、spring-mvc、spring-mybatis、spring-datasource、spring-cacher-redis、spring-cacher-ehcache等xml
2、增加目录org.cobard.config以及目录下javaconfig文件

Commit:
javaconfig替代xml配置

yzhang921 commented:
怎么切换CAS和JDBC呢

Zhazhanan commented:
![sample](https://user-images.githubusercontent.com/17845256/43379079-372d42ec-93fd-11e8-8426-666c40c62914.jpg)
![sample2](https://user-images.githubusercontent.com/17845256/43379169-ab67b5d4-93fd-11e8-8892-5e87b4b48ef6.jpg)

spring-security-cas.xml与spring-security-jdbc.xml全部用javaconfig实现，然后使用哪个就在类RootConfig.java引用哪个，目前只实现了security-jdbc方式

yzhang921 commented:
原mybatis加载properties属性的配置
spring-mabatis.xml
```xml
    <bean id="sqlSessionFactory" class="org.mybatis.spring.SqlSessionFactoryBean">
        <property name="dataSource" ref="dataSource"/>
        <property name="mapperLocations" value="classpath*:mapper/*.xml"/>
        <property name="configurationProperties">
            <util:properties location="classpath:config.properties"/>
        </property>
    </bean>
```

yzhang921 commented:
测试完成之后, 通知我一声

Commit:
javaconfig

Commit:
javaconfig optimize

Commit:
javaconfig
修改mvc消息适配方式

Commit:
javaconfig
移除lombok
